### PR TITLE
Use template strings for string concatenation in JS code in dags.html

### PR DESCRIPTION
**Pull request by XD-DENG**
- use template strings over '+', for cleaner code.
- fix bugs caused by '+' concatenation (please refer to my in-line comment below.)

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/www/templates/airflow/dags.html [hunk 1]
```diff
@@ -401,15 +401,15 @@ <h2>DAGs</h2>
...
+          return `translate(${x},${y})`;
         });
 
       g.append('svg:a')
         .attr('href', function(d) {
-          return '{{ url_for('DagRunModelView.list') }}?_flt_3_dag_id=' + dag_id + '&_flt_3_state=' + d.state;
+          return `{{ url_for('DagRunModelView.list') }}?_flt_3_dag_id=${dag_id}&_flt_3_state=${d.state}`;
         })
       .append('circle')
-        .attr('id', function(d) {return 'run-' + dag_id.replace(/\./g, '_') + d.state || 'none'})
```
Comment (XD-DENG) [inline airflow/www/templates/airflow/dags.html:412]: This is the bug I'm referring to. 

When `d.state` is `null` or `undefined`, it's like:
```javascript
'a-' + null || 'none'
```

This gives us `a-null`, while we actually expect `a-none`. So this should be
```javascript
'a-' + (null || 'none')
```

But of course this is fixed if we use template string for this.

File: airflow/www/templates/airflow/dags.html [hunk 2]
```diff
@@ -475,15 +475,15 @@ <h2>DAGs</h2>
...
+          return `translate(${x},${y})`;
         });
 
       g.append('svg:a')
         .attr('href', function(d) {
-          return '{{ url_for('TaskInstanceModelView.list') }}?_flt_3_dag_id=' + dag_id + '&_flt_3_state=' + d.state;
+          return `{{ url_for('TaskInstanceModelView.list') }}?_flt_3_dag_id=${dag_id}&_flt_3_state=${d.state}`;
         })
       .append('circle')
-        .attr('id', function(d) {return 'task-' + dag_id.replace(/\./g, '_') + d.state || 'none'})
```
Comment (XD-DENG) [inline airflow/www/templates/airflow/dags.html:486]: The same issue here

**Review by ryanahamilton — APPROVED** [review]
Much appreciated cleanup @XD-DENG!

I had started on integrating/enforcing ESLint that would ensure this pattern is always used—I need to pick that work back up soon.

**Comment by XD-DENG** [comment]
> Much appreciated cleanup @XD-DENG!
> 
> I had started on integrating/enforcing ESLint that would ensure this pattern is always used—I need to pick that work back up soon.

Thanks @ryanahamilton :)

The Jinja template usage mixed inside may cause difficulty to automate that? Would be interested to know & learn how this will be done👍

**Pull request merged**